src/scripts/generate-test-patient-spreadsheets.sh

```diff
@@ -0,0 +1,186 @@
+  log "INFO" "Generating curled test data spreadsheet for ${TEMPLATE_FILE} and outputting to ${outputXlsx}"
+  trap "onExit" EXIT
+
+  if [ ! -f "${TEMPLATE_FILE}" ]; then
+    log "ERROR" "${TEMPLATE_FILE} does not exist."
+    return 1
+  fi
+
+  if ! jq -e . "${TEMPLATE_FILE}" >/dev/null 2>&1; then
+    log "ERROR" "${TEMPLATE_FILE} is not valid json."
+    return 1
+  fi
+
+  if ! command -v libreoffice &> /dev/null; then
+    log "ERROR" "libreoffice is not installed. Please install libreoffice to use this script."
+    exit 1
+  fi
+
+  generateCsv "${tmpOutputCsv}"
+
+  convertToXlsx "${tmpOutputCsv}" "${outputXlsx}"
+
+  log "INFO" "Done"
+}
+
+log () { echo "$(date --utc +%FT%TZ) [${1}] ${2}"; }
+
+onExit() {
+  rm -rf "$WORK"
+}
+
+request() {
+  local url="${1}" curlResponse="${2}" statusCode
+
+  log "INFO" "Requesting ${url}"
+  if [ "200" != "$(curl -s -o "${curlResponse}" -w "%{http_code}" -H "Authorization: Bearer $TOKEN" "${url}")" ]; then
+    log "INFO" "Refreshing token and retrying ${url}"
+    TOKEN=$(new-token)
+    statusCode=$(curl -s -o "${curlResponse}" -w "%{http_code}" -H "Authorization: Bearer $TOKEN" "${url}")
+    if [ "200" != "${statusCode}" ]; then
+      log "ERROR" "Request to ${url} failed. Status code was ${statusCode}."
+      exit 1
+    fi
+  fi
+}
+
+generateCsv() {
+  local outputCsv="${1}"
+  local curlResponse baseUrl patientFields resourceStaticFields resourceJqFields patientFieldValues staticFieldValues
+  curlResponse=$(mktemp -p "${WORK:-/tmp}")
+  baseUrl=$(jq -r '.baseUrl' "${TEMPLATE_FILE}")
+
+  patientFields=$(jq -r '.patientFields[].name' "${TEMPLATE_FILE}" | tr '\n' ' ')
+  resourceStaticFields=()
+  resourceJqFields=()
+
+  for i in $(seq 0 $(($(jq '.resources | length' "${TEMPLATE_FILE}")-1))); do
+    for staticField in $(jq -r ".resources[${i}].staticFields[].name" "${TEMPLATE_FILE}"); do
+      if [[ ! " ${resourceStaticFields[*]} " =~ [[:space:]]${staticField}[[:space:]] ]]; then
+          resourceStaticFields+=" ${staticField}"
+      fi
+    done
+    for jqField in $(jq -r ".resources[${i}].jqFields[].name" "${TEMPLATE_FILE}"); do
+      if [[ ! " ${resourceJqFields[*]} " =~ [[:space:]]${jqField}[[:space:]] ]]; then
+          resourceJqFields+=" ${jqField}"
+      fi
+    done
+  done
+
+  echo "ICN,${patientFields// /,}Resource${resourceStaticFields// /,}${resourceJqFields// /,}" > "${outputCsv}"
+
+
+  TOKEN=$(new-token)
+
+  # Send the requests and create the spreadsheet rows
+  for patientId in $(jq -r '.testPatients[]' "${TEMPLATE_FILE}"); do
+    request "${baseUrl}/Patient/${patientId}" "${curlResponse}"
+    patientFieldValues="${patientId}"
+    for field in ${patientFields}; do
+      patientFieldValues+=","
+      value=$(jq -r "$(jq -r ".patientFields[] | select(.name==\"${field}\") | .path" "${TEMPLATE_FILE}")" "${curlResponse}")
+      if [ "${value}" != "null" ]; then
+        patientFieldValues+="${value//,/ }"  # Replace commas with spaces
```

Comment:
Should we replace it with spaces or some other character?

Reply:
Do you have another character in mind?

Reply:
For CSV parsing, it would just be surrounded by `"`. I would see if libreoffice can deal with that, but I almost was thinking encode it otherwise. That way theres no data loss, so to speak.